src/web/crate_details.rs

```diff
@@ -91,6 +92,34 @@ pub struct Release {
     pub target_name: String,
 }
 
+#[fn_error_context::context("fetching readme for {name} {version}")]
+fn fetch_readme_from_source(
+    storage: &Storage,
+    name: &str,
+    version: &str,
+    archive_storage: bool,
+) -> anyhow::Result<String> {
+    let manifest = storage.fetch_source_file(name, version, "Cargo.toml", archive_storage)?;
+    let manifest = String::from_utf8(manifest.content)
+        .context("parsing Cargo.toml")?
+        .parse::<toml::Value>()
+        .context("parsing Cargo.toml")?;
+    let paths = match manifest.get("package").and_then(|p| p.get("readme")) {
+        Some(toml::Value::Boolean(true)) => vec!["README.md"],
+        Some(toml::Value::Boolean(false)) => vec![],
+        Some(toml::Value::String(path)) => vec![path.as_ref()],
+        _ => vec!["README.md", "README.txt", "README"],
```

Comment:
isn't this logic that we would also find somewhere in the `cargo` codebase? or some shared library?

Reply:
https://github.com/rust-lang/cargo/blob/9a7c1fe322a9c4494ffd2c2684626964d1b06c83/src/cargo/util/toml/mod.rs#L2992

`cargo` is a massive library dependency we don't want to pull in though, I don't think there's any smaller libraries implementing this part of it.